android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/views/DynamicallyOrientingNavigationView.kt

```diff
@@ -22,49 +34,68 @@ import com.stadiamaps.ferrostar.maplibreui.runtime.navigationMapViewCamera
  * @param locationRequestProperties The location request properties to use for the map's location
  *   engine.
+ * @param config The configuration for the navigation view.
  * @param content Any additional composable map symbol content to render.
- * @param orientation The orientation of the device. Defaults to the current device orientation, but
- *   can be overridden (e.g. with Configuration.ORIENTATION_LANDSCAPE or
- *   Configuration.ORIENTATION_PORTRAIT).
  */
 @Composable
 fun DynamicallyOrientingNavigationView(
     modifier: Modifier,
-    orientation: Int = LocalConfiguration.current.orientation,
     styleUrl: String,
     camera: MutableState<MapViewCamera> = rememberSaveableMapViewCamera(),
     navigationCamera: MapViewCamera = navigationMapViewCamera(),
     viewModel: NavigationViewModel,
     locationRequestProperties: LocationRequestProperties =
         LocationRequestProperties.NavigationDefault(),
+    config: VisualNavigationViewConfig = VisualNavigationViewConfig.Default(),
     onTapExit: (() -> Unit)? = null,
     content: @Composable @MapLibreComposable() ((State<NavigationUiState>) -> Unit)? = null,
 ) {
-  when (orientation) {
-    Configuration.ORIENTATION_LANDSCAPE -> {
-      LandscapeNavigationView(
-          modifier,
-          styleUrl,
-          camera,
-          navigationCamera,
-          viewModel,
-          locationRequestProperties,
-          onTapExit,
-          content)
-    }
-    else -> {
-      PortraitNavigationView(
-          modifier,
-          styleUrl,
-          camera,
-          navigationCamera,
-          viewModel,
-          locationRequestProperties,
-          onTapExit,
-          content)
+  SystemUIDisposableEffect()
+
+  val orientation = LocalConfiguration.current.orientation
+  val isLandscape = orientation == Configuration.ORIENTATION_LANDSCAPE
+
+  Box(modifier) {
+    NavigationMapView(
+        styleUrl,
+        mapControlsFor(isLandscape = isLandscape, isArrivalExpanded = onTapExit != null),
+        camera,
+        navigationCamera,
+        viewModel,
+        locationRequestProperties,
+        onMapReadyCallback = { camera.value = navigationCamera },
+        content)
+
+    when (orientation) {
+      Configuration.ORIENTATION_LANDSCAPE -> {
+        LandscapeNavigationOverlayView(
+            modifier =
+                Modifier.fillMaxSize()
+                    // TODO: DO NOT MERGE - Finalize w/ device
+                    .padding(top = 32.dp, start = 16.dp, end = 16.dp, bottom = 16.dp),
+            camera = camera,
+            viewModel = viewModel,
+            config = config,
+            onTapExit = onTapExit)
+      }
+      else -> {
+        PortraitNavigationOverlayView(
+            modifier =
+                Modifier.fillMaxSize()
+                    .statusBarsPadding()
+                    .navigationBarsPadding()
+                    .padding(horizontal = 16.dp)
+                    .padding(bottom = 24.dp),
```

Comment:
Assume this is similar to the above. Seems like refactoring these as constants are the least we should do just so it's obvious what's going on. Might turn into something more elaborate like an interface later.